### DISCUSSION TITLE
Text element that is divided between two pages

### DISCUSSION THREAD

**Discussion by 838** [readium/architecture#178] [Q&A] [answered]
In our app we try to highlight the current narrated paragraph and sometimes the paragraph breaks into the next page which this causes an issue with the audiobooks that the page will not be updated until the audio reaches the next paragraph.

Any recommendation of how we could solve this issue? 

https://user-images.githubusercontent.com/3314171/177775877-49216b93-d743-4154-b163-58960fbfb534.mov

**Comment by mickael-menu** [comment, marked as answer]
What technology are you using for the playback? Is it TTS? Media overlays? Did you implement a custom solution?

**Reply by 838** [reply]
We are building a Flutter app that wraps Readium's native Navigator(swift&kotling and js in future) with our custom highlighting solution that wraps the narrated text with a `span` element.

When the duration of audio updates we calculate which element should be highlighted by finding the media over from the player duration, but the issue is that sometimes based on font/screen size a paragraph text could be divided between two pages and we try to find a solution to jump to the next page when the audio reaches the text at the end of the page (Fx. "til den kniv" from the demo video).

**Reply by mickael-menu** [reply]
What's the source of the audio? An audio file in the publication or are you synthesizing the raw text with a TTS engine?

**Reply by 838** [reply]
An audio file in the publication.

**Reply by mickael-menu** [reply]
Then I'm afraid it's not possible, unless you have a timestamp for each word being read in your narrated synchronization manifest.

If you manage to get a word-by-word event, you could use `navigator.go(to: Locator)` using a `Locator` with the current word in `text.highlight` and the previous and next text content in `text.before` and `text.after`.

**Reply by 838** [reply]
@mickael-menu Thank you for your useful feedback 🙏🙂.
